docs/design/integrate-with-cc-tools/README.md

```diff
@@ -0,0 +1,300 @@
+
+1. Stub interface order:
+
+    cc-tools is a package that provides a relational-like framework for programming fabric chaincodes and it translates every code to a normal fabric chaincode at the end. cc-tools is wrapping the [shim.ChaincodeStub](https://github.com/hyperledger/fabric-chaincode-go/blob/acf92c9984733fb937fba943fbf7397d54368751/shim/interfaces.go#L28) interface from Hyperledger Fabric using [stubWrapper](https://github.com/hyperledger-labs/cc-tools/blob/995dfb2a16decae95a9dbf05424819a1df19abee/stubwrapper/stubWrapper.go#L12) and if you look for example at the [PutState](https://github.com/hyperledger-labs/cc-tools/blob/995dfb2a16decae95a9dbf05424819a1df19abee/stubwrapper/stubWrapper.go#L18) function you notice it only does some in-memory operations and it calls another `sw.Stub.PutState` from the stub passed to it (till now it was always the [stub for standard fabric](https://github.com/hyperledger/fabric-chaincode-go/blob/main/shim/stub.go)).
+
+    ```go
+    package stubwrapper
+
+    import (
+        "github.com/hyperledger-labs/cc-tools/errors"
+        "github.com/hyperledger/fabric-chaincode-go/shim"
+    )
+
+    type StubWrapper struct {
+        Stub        shim.ChaincodeStubInterface
+        WriteSet    map[string][]byte
+        PvtWriteSet map[string]map[string][]byte
+    }
+
+    func (sw *StubWrapper) PutState(key string, obj []byte) errors.ICCError {
+        err := sw.Stub.PutState(key, obj)
+        if err != nil {
+            return errors.WrapError(err, "stub.PutState call error")
+        }
+
+        if sw.WriteSet == nil {
+            sw.WriteSet = make(map[string][]byte)
+        }
+        sw.WriteSet[key] = obj
+
+        return nil
+    }
+
+    ```
+
+    On the other hand for FPC, it also wraps the [shim.ChaincodeStub](https://github.com/hyperledger/fabric-chaincode-go/blob/acf92c9984733fb937fba943fbf7397d54368751/shim/interfaces.go#L28) interface from Hyperledger Fabric but the wrapper it uses (in this case it's [FpcStubInterface](https://github.com/hyperledger/fabric-private-chaincode/blob/33fd56faf886d88a5e5f9a7dba15d8d02d739e92/ecc_go/chaincode/enclave_go/shim.go#L17)) is not always using the functions from the passed stub. With the same example as before, if you look at the [PutState](https://github.com/hyperledger/fabric-private-chaincode/blob/33fd56faf886d88a5e5f9a7dba15d8d02d739e92/ecc_go/chaincode/enclave_go/shim.go#L104) function you can notice it's not using the `sw.Stub.PutState` and going directly to the `rwset.AddWrite` (this is specific to fpc use case as it's not using the fabric proposal response). There are some other functions where the passed `stub` functions are being used.
```

Comment:
There is a FPC that is lower case ... you should have the same case .... it says "this is specific to fpc use case" also in the following sentence you want to say that in FPC there are some functions that call the stub functions and not something else like you explained before? If so, then it should be "There still are other functions where the ....."

Reply:
Done